examples/chicago_taxi/tfdv_analyze_and_validate_flink.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -u
+
+echo Starting distributed TFDV stats computation and schema generation...
+
+DATA_DIR=$(pwd)/data
+OUTPUT_DIR=$DATA_DIR/flink_tfdv_output
+
+JOB_ID="chicago-taxi-tfdv-$(date +%Y%m%d-%H%M%S)"
+JOB_OUTPUT_PATH=$OUTPUT_DIR
+
+# Variables needed for subsequent stages.
+TFDV_OUTPUT_PATH=$JOB_OUTPUT_PATH
+SCHEMA_PATH=$TFDV_OUTPUT_PATH/schema.pbtxt
+
+echo Job output path: $JOB_OUTPUT_PATH
+echo TFDV output path: $TFDV_OUTPUT_PATH
+
+mkdir -p data/flink_tfdv_output
+
+EXTRA_ARGS="--infer_schema \
+            --stats_path $TFDV_OUTPUT_PATH/train_stats.tfrecord \
+            --schema_path $SCHEMA_PATH \
+            --save_main_session True \
+            --input $DATA_DIR/train/data.csv " SCRIPT=tfdv_analyze_and_validate.py $(pwd)/execute_on_flink.sh
+
+EXTRA_ARGS="--for_eval \
+            --validate_stats \
+            --stats_path $TFDV_OUTPUT_PATH/eval_stats.tfrecord \
+            --schema_path $SCHEMA_PATH \
+            --anomalies_path $TFDV_OUTPUT_PATH/anomalies.pbtxt \
+            --save_main_session True \
+            --input $DATA_DIR/eval/data.csv " SCRIPT=tfdv_analyze_and_validate.py $(pwd)/execute_on_flink.sh
+
```

Comment:
so if I understand right, we can do tfdv and tft pre processing on flink runner

This is great! 

Just being nitpicky here, I wonder if we can keep existing scripts, and have arguments passed to it to a)start fliink cluster, and b) pass in the Portable runner with required arguments for that

Also do you have plans to add support for tfma, it would probably be exactly the same steps ?

Reply:
a) These scripts can use the inprocess flink in beam job sever which remove the requirement for a flink cluster. However, user has to start a beam job server. I can automate download and building of beam from git. Please suggest if I should add it in this PR.
b) Using portable runner is take care of in execute_on_flink.sh . From the perspective of example, we don't have to pass any additional arguments for portable runner.

I tried running local tfma but it failed with some module exception. I will try to fix and then can add flink tfma

Reply:
SGTM